src/libraries/Microsoft.Extensions.Hosting.Abstractions/ref/Microsoft.Extensions.Hosting.Abstractions.cs

```diff
@@ -23,6 +23,11 @@ public abstract partial class BackgroundService : Microsoft.Extensions.Hosting.I
         public virtual System.Threading.Tasks.Task StartAsync(System.Threading.CancellationToken cancellationToken) { throw null; }
         public virtual System.Threading.Tasks.Task StopAsync(System.Threading.CancellationToken cancellationToken) { throw null; }
     }
+    public enum BackgroundServiceExceptionBehavior
```

Comment:
Why are we putting the enum in `Hosting.Abstractions`, when the only usage of it is in `Hosting`?

Reply:
I was actually unsure about that, I saw other enum definitions in the abstractions and assumed that was where they belonged. I was curious about what determines where these belong, and the meaning of `ref` as it relates to `src` and `tests`. I was also thinking that is would be a really cool use-case for source generators if `ref` are just the API surface areas of the bits defined in `src`. I'll move this, thanks.